java/source/io.rst

```diff
@@ -0,0 +1,354 @@
+       metadata.put("A", "Id card");
+       metadata.put("B", "Passport");
+       metadata.put("C", "Visa");
+       Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), null, metadata), null);
+
+       Field age = new Field("age", FieldType.nullable(new ArrowType.Int(32, true)), null);
+
+       FieldType intType = new FieldType(true, new ArrowType.Int(32, true), /*dictionary=*/null);
+       FieldType listType = new FieldType(true, new ArrowType.List(), /*dictionary=*/null);
+       Field childField = new Field("intCol", intType, null);
+       List<Field> childFields = new ArrayList<>();
+       childFields.add(childField);
+       Field points = new Field("points", listType, childFields);
+
+       // create a definition
+       Schema schemaPerson = new Schema(asList(name, document, age, points));
+
+       RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE); // deal with byte buffer allocation
+       VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(schemaPerson, rootAllocator);
+
+       // getting field vectors
+       VarCharVector nameVectorOption1 = (VarCharVector) vectorSchemaRoot.getVector("name"); //interface FieldVector
+       VarCharVector documentVectorOption1 = (VarCharVector) vectorSchemaRoot.getVector("document"); //interface FieldVector
+       IntVector ageVectorOption1 = (IntVector) vectorSchemaRoot.getVector("age");
+       ListVector pointsVectorOption1 = (ListVector) vectorSchemaRoot.getVector("points");
+
+       // add values to the field vectors
+       setVector(nameVectorOption1, "david".getBytes(), "gladis".getBytes(), "juan".getBytes());
+       setVector(documentVectorOption1, "A".getBytes(), "B".getBytes(), "C".getBytes());
+       setVector(ageVectorOption1, 10,20,30);
+       setVector(pointsVectorOption1, asList(1,3,5,7,9), asList(2,4,6,8,10), asList(1,2,3,5,8));
+       vectorSchemaRoot.setRowCount(3);
+
+       return vectorSchemaRoot;
+   }
+
+   RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE); // deal with byte buffer allocation
+
+   VectorSchemaRoot vectorSchemaRoot = createVectorSchemaRoot();
+
+
+.. code-block:: java
+   :emphasize-lines: 1-6
+
+   jshell> System.out.println(vectorSchemaRoot.contentToTSVString())
+
+   name     document age   points
+   david    A        10    [1,3,5,7,9]
+   gladis   B        20    [2,4,6,8,10]
+   juan     C        30    [1,2,3,5,8]
+
+Writing arrays with the IPC file format
+***************************************
+
+Write - Random access to file
+-----------------------------
+
+.. code-block:: java
+   :emphasize-lines: 9
+
+   import org.apache.arrow.vector.ipc.*;
+
+   import java.io.*;
+
+   // random access format
+   // write - random access to file
+   File file = new File("randon_access.arrow");
+   FileOutputStream fileOutputStream = new FileOutputStream(file);
+   ArrowFileWriter writer = new ArrowFileWriter(vectorSchemaRoot, null, fileOutputStream.getChannel());
```

Comment:
Don't we need to close the writer, file, etc.?

Reply:
We decided to move all of this best practices to [java tutorial ticket](https://issues.apache.org/jira/browse/ARROW-15156)